app/src/components/SyncService.js

```diff
@@ -94,6 +99,62 @@ const SyncService = () => {
     };
   }, [syncInSecond, isOnline, onSync]);
 
+  const onSyncDataPoint = useCallback(async () => {
+    const activeJob = await crudJobs.getActiveJob(SYNC_DATAPOINT_JOB_NAME);
+
+    if (activeJob.status === PENDING && activeJob.attempt < 3) {
+      UIState.update((s) => {
+        s.statusBar = {
+          type: syncStatus.ON_PROGRESS,
+          bgColor: '#2563eb',
+          icon: 'sync',
+        };
+      });
+      await crudJobs.updateJob(activeJob.id, {
+        status: jobStatus.ON_PROGRESS,
+      });
+      try {
+        const allData = await fetchDatapoints(activeJob.form);
+        const urls = allData.map((item) => item.url);
+        await Promise.all(urls.map(downloadDatapointsJson));
+        UIState.update((s) => {
+          s.statusBar = {
+            type: syncStatus.SUCCESS,
+            bgColor: '#16a34a',
+            icon: 'checkmark-done',
+          };
+        });
+        await crudJobs.deleteJob(activeJob.id);
+      } catch (error) {
+        await crudJobs.updateJob(activeJob.id, {
+          status: jobStatus.PENDING,
+          attempt: activeJob.attempt + 1,
+        });
+      }
+    }
+
+    if (activeJob.status === PENDING && activeJob.attempt === 3) {
+      await crudJobs.deleteJob(activeJob.id);
+    }
+
+    console.log(activeJob, 'activeJob');
+  }, [statusBar]);
+
+  useEffect(() => {
+    if (!syncInSecond || !isOnline) {
+      return;
+    }
+    const syncTimer = setInterval(() => {
+      // Perform sync operation
+      onSyncDataPoint();
+    }, syncInSecond);
+
+    return () => {
+      // Clear the interval when the component unmounts
+      clearInterval(syncTimer);
+    };
+  }, [syncInSecond, isOnline, onSync]);
```

Comment:
remove syncInSecond from this useEffect dependencies, Because the interval is static 1 sec, not from settings.

```
const jobInterval = 1 * 1000; // 1 second
if (!isOnline) {
  return;
}
const syncTimer = setInterval(() => {
  // Perform sync operation
  onSyncDataPoint();
}, jobInterval);
```